pandas/core/reshape/pivot.py

```diff
@@ -611,13 +611,16 @@ def _normalize(table, normalize, margins, margins_name="All"):
         table = table.fillna(0)
 
     elif margins is True:
+        # keep index and column of pivoted table
+        table_index = table.index
+        table_columns = table.columns
 
-        column_margin = table.loc[:, margins_name].drop(margins_name)
-        index_margin = table.loc[margins_name, :].drop(margins_name)
-        table = table.drop(margins_name, axis=1).drop(margins_name)
-        # to keep index and columns names
-        table_index_names = table.index.names
-        table_columns_names = table.columns.names
+        # save the column and index margin
```

Comment:
if margins is True, then we are guaranteed to have the margins_name be the last row / column? can you add an assert to this, that the last row.name / col.name == margins_name

Reply:
yes, since this normalization will take one step further from the output of `pivot_table` function, and in this function, if margin is set to `True`, then there will be a new `column/index` added to the end (`'All' or 'New_Margin_Name'`). But you are right, it's better to add an assertion to this, will do later today! @jreback

Reply:
great, this just makes it clear to a future reader, otherwise lgtm. ping on green.

Reply:
I have added assertion, pls feel free to take a look @jreback